src/features/customerView/utils.ts

```diff
@@ -140,3 +141,78 @@ export const getApplications = (profile: CUSTOMER_PROFILE, boatTypes: BOAT_TYPES
     }, []) ?? []
   );
 };
+
+type OrderLine = {
+  product: ProductServiceType;
+  price: number;
+  taxPercentage: number;
+};
+
+type Bill = {
+  contractPeriod: {
+    startDate: string;
+    endDate: string;
+  };
+  dueDate: string;
+  totalPrice: number;
+  totalPriceTaxPercentage: number;
+  basePrice: number;
+  basePriceTaxPercentage: number;
+  orderLines: OrderLine[];
+};
+
+export type BerthBill = Bill & {
+  berthInformation: {
+    number: number;
+    pierIdentifier: string;
+    harborName: string;
+  };
+};
+
+export const getOpenBills = (profile: CUSTOMER_PROFILE): BerthBill[] => {
+  return (
+    profile.berthLeases?.edges
+      .map((edge) => edge?.node)
+      .filter((node) => node?.order?.status === OrderStatus.WAITING)
+      .reduce<BerthBill[]>((acc, berthLeaseNode) => {
```

Comment:
Since we're calling reduce here, I think all of these three operations can be achieved in `reduce`'s callback

Reply:
The filter has now been removed and the function just gets all bills.